### PR TITLE
test(hc): Change default test behavior to silo modes

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Have all test cases tagged with a silo mode decorator (`@control_silo_test` or `@region_silo_test`) run only in that mode. The `stable` argument is now true by default, and the test is not run in monolith mode unless it has `include_monolith_run=True`.

### CODE REVIEW

**Comment by corps** [comment]
Sweet!

**Review by mrduncan — APPROVED** [review]
🚀

File: tests/sentry/api/endpoints/test_user_authenticator_enroll.py [hunk 1]
```diff
@@ -268,7 +268,7 @@ def test_u2f_disallow_new_enrollment(self):
         )
 
 
-@control_silo_test(stable=True)
+@control_silo_test(stable=True, include_monolith_run=True)
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_user_authenticator_enroll.py:271]: :clap: This is a nice way to retain coverage for the small number of monolith paths that we have.

**Review by markstory — COMMENTED** [review]
As a follow up to this it would be good to remove the `stable=True` parameter.

**Comment by RyanSkonnord** [comment]
Blocked by https://github.com/getsentry/sentry/pull/60331, which fixes a test case bug that this change exposes.

**Review by GabeVillalobos — APPROVED** [review]

**Pull request merged**